M/mlpack/build_tarballs.jl

```diff
@@ -185,9 +188,10 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("boost_jll"; compat="=1.76.0"),
     Dependency("armadillo_jll"),
-    Dependency("OpenBLAS_jll", v"0.3.13")
+    Dependency("OpenBLAS_jll", v"0.3.13"),
```

Comment:
From the audit https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=23225&view=logs&j=f5128b8a-f209-5fa4-85df-c01db7529e06&t=d51dec25-fb1c-5234-c33e-37032ff96d86&l=15729:
```
┌ Warning: To ensure that the correct version of libgomp is found at runtime, add the following entry to the list of dependencies of this builder
│ 
│     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/xmfIp/src/auditor/compiler_abi.jl:4
```
```suggestion
    Dependency("OpenBLAS_jll", v"0.3.13"),
    Dependency("CompilerSupportLibraries_jll"),
```